src/RulesDeploy.ts

```diff
@@ -33,46 +38,46 @@ RulesDeploy.prototype = {
     }
 
     this.rulesFiles[path] = [{ name: path, content: src }];
-  },
+  }
 
   /**
    * Compile all rulesets tied to this deploy, rejecting on first
    * compilation error.
    */
-  compile: function() {
-    var self = this;
-    var promises = [];
-    _.forEach(this.rulesFiles, function(files, filename) {
+  compile(): Promise<any> {
+    const self = this;
```

Comment:
Throughout the file: `self` can be removed (and uses of `self` can be replaced with `this`). The references should all work fine.

Reply:
done